src/commands/tags/registerTagDiagnostics.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Diagnostic, languages, TextDocument, TextDocumentChangeEvent, TextEditor, window, workspace } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext, registerEvent } from "vscode-azureextensionui";
+import { ext } from "../../extensionVariables";
+import { getTagDiagnostics } from "./getTagDiagnostics";
+import { TagFileSystem } from "./TagFileSystem";
+
+export function registerTagDiagnostics(): void {
+    ext.diagnosticCollection = languages.createDiagnosticCollection('Azure Tags');
+    ext.context.subscriptions.push(ext.diagnosticCollection);
+
+    registerEvent('onDidChangeActiveTextEditor', window.onDidChangeActiveTextEditor, onDidChangeActiveTextEditor);
+}
+
+function onDidChangeActiveTextEditor(context: IActionContext, editor: TextEditor | undefined): void {
+    context.telemetry.suppressIfSuccessful = true;
+    context.errorHandling.suppressDisplay = true;
+
+    if (editor?.document.uri.scheme === TagFileSystem.scheme) {
+        if (!ext.diagnosticWatcher) {
+            ext.diagnosticWatcher = workspace.onDidChangeTextDocument(onDidChangeTextDocument);
+        }
+        updateTagDiagnostics(editor.document);
+    } else if (ext.diagnosticWatcher) {
```

Comment:
How/When does this get created/set?  I guess you're clearing it when the scheme (which I assume is azureResourceGroup) doesn't match?  So you dispose it anytime the person opens a file that is not opened from the extension?

Reply:
It's set here: https://github.com/microsoft/vscode-azureresourcegroups/pull/15/files#diff-852c7ed0a0de11b8c9f3efd1029cd634R25

It's really just a disposable for the `onDidChangeTextDocument` event, which is fired _every time the user types something_. I set it when they open a tags file and dispose it when they open a different file. Just trying to make sure we don't hurt the performance of normal files